File: 2015/06/2.rb
lights = Hash.new(0)
regexp = /(turn on|turn off|toggle) (\d+,\d+) through (\d+,\d+)/

def coordinates(start, finish)
  Enumerator.new do |yielder|
    for x in start.first..finish.first do
      for y in start.last..finish.last do
        yielder << "#{x},#{y}"
      end
    end
  end
end

File.open('input.txt').each do |line|
  instruction, start, finish = line.match(regexp).captures
  start = start.split(',').map(&:to_i)
  finish = finish.split(',').map(&:to_i)

  case instruction
  when 'turn on'
    for xy in coordinates(start, finish) do
      lights[xy] += 1
    end
  when 'turn off'
    for xy in coordinates(start, finish) do
      lights[xy] -= 1 if lights[xy] > 0
    end
  when 'toggle'
    for xy in coordinates(start, finish) do
      lights[xy] += 2
    end
  end
end

puts lights.values.sum

File: 2022/02/2.py
round = {'X': 0, 'Y': 3, 'Z': 6}
score = 0

with open("input.txt") as f:
    for line in f:
        opponent, player = line.rstrip().split(' ')
        match (opponent, player):
            case ('A', 'X') | ('B', 'Z') | ('C', 'Y'):
                score += 3
            case ('A', 'Z') | ('B', 'Y') | ('C', 'X'):
                score += 2
            case ('A', 'Y') | ('B', 'X') | ('C', 'Z'):
                score += 1
        score += round[player]

print(score)
